src/MARIGOLD/iate.py

```diff
@@ -0,0 +1,364 @@
+    jgloc           = cond.jgloc                                # [m/s]
+
+    if 'jgatm' in dir(cond):
+        jgatm       = cond.jgatm
+    else:
+        print("Warning: jgatm not found. Defaulting to jgref")
+        jgatm       = cond.jgref
+    
+    p               = (jgatm * p_atm / jgloc) - p_atm           # Back-calculate local corrected gauge pressure
+
+	# Local Pressure along the test section
+    omegaz = 1 - (z_mesh - z_mesh[0]) * abs(dpdz / (p + p_atm))
+    pz = (p + p_atm) * omegaz
+    
+	# Local gas density along the test section
+    rho_gz = rho_g * pz / p_atm                                 # Talley
+    # rho_gz = pz / R_spec / T                                  # Worosz, Ideal Gas Law
+    
+    ############################################################################################################################
+    #                                                                                                                          #
+    #                                                           IATE                                                           #
+    #                                                                                                                          #
+    ############################################################################################################################
+    # Calculate ai(z) to evaluate the steady state one-dim one-group model
+
+    for i, z in enumerate(z_mesh):
+        if (i+1) >= len(z_mesh):
+            break
+            
+        jgloc = jgatm * p_atm / pz[i]                           # Talley used jgP1 and pressure at P1 instead of jgatm and p_atm
+
+        vgz[i] = jgloc / alpha[i]                               # Estimate void weighted velocity
+        vfz = jf / (1 - alpha[i])
+
+        ########################################################################################################################
+        # Estimate bubble relative velocity <ur> (See Talley, 2012, 4.2.2.6)
+        ur = 0.2        # Set to constant value of 0.23 m/s by Schilling (2007)
+        '''
+        err = 0.1
+        while abs(err) > 0.000001:
+            ReD = rho_f * ur * Db[i] * (1 - alpha[i]) / mu_f    # Bubble Reynolds number
+            CDe = 24 * (1 + 0.1 * ReD**0.75) / ReD              # Drag coefficient
+
+            # Relative velocity (Ishii and Chawla, 1979), implemented by Worosz accounting for density difference
+            ure1 = (4 * grav * Db[i] * (rho_f - rho_gz[i]) * (1 - alpha[i]) / 3 / CDe / rho_f)**0.5
+            err = (ure1 - ur) / ur
+
+            ur = ure1
+        ReD = rho_f * ur * Db[i] * (1 - alpha[i]) / mu_f        # Update bubble Reynolds number
+        CDwe = 24 * (1 + 0.1 * ReD**0.75) / ReD                 # Update drag coefficient
+        '''
+
+        # Original DOE_MATLAB_IAC
+        ReD = rho_f * ur * Db[i] * (1 - alpha[i]) / mu_f
+        CDwe = 24 * (1 + 0.1 * ReD**0.75) / ReD
```

Comment:
Could use cond.calc_CD method here

Reply:
That way it'll interface with ur and such

Reply:
I'm not so sure; this calculation occurs now during the looping part of the script as it marches along the z-axis

I think cond.calc_cd, cond.calc_mu_eff, etc. use local probe data which we don't have access to as we depart from the initial boundary condition --

They'd need to take variable area-averaged or void-area-averaged parameters as inputs, but I don't think that's really what these functions are made for

Less predicting CD or mu_eff at specific L/Ds, more finding them at whatever port we have data?

Correct me if I'm wrong